core/scripts/PrecisionErrors.js

```diff
@@ -675,89 +691,192 @@ async function runExport() {
-  breakdown.tokensOutstanding = new CollateralBreakdown(tokensOutstanding);
-  breakdown.feeMultiplier = new CollateralBreakdown(actualFeeMultiplier);
+  breakdown["Collateral owned by Contract"] = new CollateralBreakdown(contractCollateral);
+  breakdown["Sponsor tokens outstanding"] = new CollateralBreakdown(tokensOutstanding);
   console.group(`** After 1 Redemption of ${fromWei(testConfig.amountToRedeem)} collateral: **`);
   console.table(breakdown);
   console.groupEnd();
 
   /**
-   * @notice POST-TEST INVARIANTS
+   * @notice POST-TEST CLEANUP
    */
-  endingStoreCollateral = await collateral.balanceOf(store.address);
 
-  // Test 1) Make sure that store hasn't collected any fees during this test, so that we can be confident that withdraws
-  // are the only source of drift.
-  assert.equal(startingStoreCollateral.toString(), endingStoreCollateral.toString());
+  console.groupEnd();
+  /** ***************************************************************************
+   *
+   * END REDEEM()
+   *
+   *****************************************************************************/
 
-  // Test 2) The fee multiplier has not changed.
-  assert.equal(actualFeeMultiplier.toString(), testConfig.expectedFeeMultiplier);
+  /** ***************************************************************************
+   *
+   * START SETTLE_EXPIRED()
+   *
+   *****************************************************************************/
+  // Overview:
+  // - `T` = Contract's synthetic tokens outstanding
+  // - `RC` = raw collateral
+  // - `M` = fee multiplier
+  // - `C` = Collateral credited to sponsors `(RC * M)`
+  // - `E` = Settlement price (amount of collateral that each synthetic can be redeemed for)
+  // - Settlements at Expiry have two sources of precision loss:
+  //   (1) The user burns S synthetic tokens and should receive `S * E = c` collateral tokens.
+  //       It is possible that `S * E` gets floored and loses 1e-18 of precision,
+  //       therefore the proportion of collateral returned is less than the synthetic burned: `(C - c)/C < (T-S)/T`.
+  //   (2) Identically to a withdraw, `c` collateral is sent from the contract to the user,
+  //       while `(c / M)` raw collateral is removed from `RC`. `(c / M)` is floored and therefore
+  //       the contract loses more collateral than it debits from sponsors: `(RC - c/M) * M > (C - c)`.
+  // Error Amount:
+  // -1e-18 in `c` if `(S * E)` has -1e-18 error
+  // +1e-18 in `RC` for the same reason as in a withdraw.
+  // Here's an example of (1):
+  // - M = cumulativeFeeMultiplier = 1
+  // - RC = rawCollateral = 1e-18
+  // - T = tokensOutstanding = 1e-18
+  // - S = synthetic balance to settle = 1e-18
+  // - E = settlement price (collateral per synthetic) = 0.5
+  // - (S * E) = 0.5e-18, which gets floored to 0
+  // - So, the user burns 1e-18 synthetic but receives 0 collateral, representing 0.5e-18 less than they were expecting.
```

Comment:
Just need to read and understand this comment to understand the test as the structure is the same: User burns 1e-18 tokens, expects to receive 0.5e-18 collateral, which gets floored. User is rekt. 

Sponsor has 0 tokens, but their "excess collateral" is `totalContractCollateral - (totalTokensOutstanding * settlementPrice)" which is `1e-18 - (1e-18 * 0.5)` but this product again gets floored, so the sponsor receives `1e-18` tokens even though there was no "excess collateral"! Moon!

Reply:
While this results in the user losing all their funds and the sponsor gaining everything it is only because you are at the macro scale where the rounding results in all collateral being taken. In the more general case with realistic balances this will only result in losses of single least significant digit in the worst case, right?

Reply:
Yes, only 1e-18 of loss can occur on each `(S*E)` calculation